R/tcrDist3Utils.R

```diff
@@ -349,3 +349,221 @@ FormatMetadataForTcrDist3 <- function(metadata,
     stop("tcrdist3_gene_segments.txt generation failed. Check Python script execution.")
   }
 }
+
+#' Create a ComplexHeatmap for a single TCR distance assay
+#'
+#' @param seuratObj_TCR A Seurat object containing TCR distance assays.
+#' @param assay Character string specifying the assay name to plot.
+#' @param cluster_info Factor vector of cluster assignments for each cell in the assay.
+#' @param cluster_colors Named character vector of colors corresponding to cluster levels.
+#' @return A ComplexHeatmap object ready for drawing.
+#' @importFrom ComplexHeatmap Heatmap draw HeatmapAnnotation rowAnnotation
+#' @keywords internal
+.TCRDistanceHeatmap <- function(seuratObj_TCR, assay, cluster_info, cluster_colors) {
+  m <- as.matrix(Seurat::GetAssayData(seuratObj_TCR, assay = assay, layer = "counts"))
+  cluster_info <- factor(cluster_info)
+  colnames(m) <- as.character(cluster_info)
+  rownames(m) <- as.character(cluster_info)
+  
+  col_annotation <- HeatmapAnnotation(
+    cluster = cluster_info,
+    col = list(cluster = cluster_colors),
+    show_annotation_name = FALSE,
+    annotation_name_side = "left",
+    which = "column",
+    show_legend = TRUE
+  )
+  
+  row_annotation <- rowAnnotation(
+    cluster = cluster_info,
+    col     = list(cluster = cluster_colors),
+    show_annotation_name = FALSE,
+    show_legend          = FALSE
+  )
+  
+  Heatmap(
+    m,
+    name               = assay,
+    column_title       = assay,
+    border_gp          = gpar(col = "black", lty = 2),
+    row_names_gp       = grid::gpar(fontsize = 8),
+    column_names_gp    = grid::gpar(fontsize = 8),
+    top_annotation     = col_annotation,
+    left_annotation    = row_annotation,
+    use_raster         = TRUE,
+    cluster_columns    = TRUE,
+    cluster_rows       = TRUE,
+    row_dend_side      = "left",
+    column_dend_side   = "top",
+    column_split       = cluster_info,
```

Comment:
I think it'd be useful to return a heatmap without the clustering info too. Does this function break if you set `cluster_info = NULL`, or does it give you an unsplit plot?

Reply:
it could be easy to add a `annotateClusters` argument to `TCRDistanceHeatmaps()` that nests all of this: https://github.com/GWMcElfresh/tcrClustR/pull/9/files#diff-82068ca07cd61c8a971adc091b10018378a4dc19a41224ad1b4a73b479027879R436